### ISSUE TITLE
valid-jsdoc respect type guard

### ISSUE THREAD

**Issue by Shinigami92** [eslint/eslint#13390]
**What rule do you want to change?**

valid-jsdoc

**Does this change cause the rule to produce more or fewer warnings?**

Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

Be less strict

**Please provide some example code that this change will affect:**

```js
/**
 * Check if string
 *
 * @param {any} value Value
 * @returns {value is string} Is string
 */
function isString(value) {
  return typeof value === 'string';
}
```

**What does the rule currently do for this code?**

Print warning for valid type guard return type definition

**What will the rule do after it's changed?**

Print less warning

**Are you willing to submit a pull request to implement this change?**

Not well experienced with code of eslint

**Comment by yeonjuan** [comment]
Hi :) @Shinigami92 
`require-jsdoc` is deprecated in eslint core since ESLint v5.10.0. - details in [here](https://eslint.org/blog/2018/11/jsdoc-end-of-life#current-jsdoc-support-in-eslint)

> The require-jsdoc and valid-jsdoc rules will be deprecated. These two rules will remain in ESLint but we will no longer add new features or fix bugs for them. These rules may be removed in a future major release of ESLint.

As described in the docs, `valid-jsdoc` will no longer be updated.
Alternatively, there is an [eslint plugin for jsdoc](https://github.com/gajus/eslint-plugin-jsdoc) (I'm not sure the plugin contains exactly same rule with `require-jsdoc`)
